//
// Copyright 2019 The Project Oak Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package oak;

// Interface exposed by the Escrow Node to other nodes over a pair of Oak Channels.
service EscrowNode {
  rpc Read(EscrowChannelReadRequest) returns (EscrowChannelReadResponse);
  rpc Write(EscrowChannelWriteRequest) returns (EscrowChannelWriteResponse);
  rpc Delete(EscrowChannelDeleteRequest) returns (EscrowChannelDeleteResponse);
}

message EscrowChannelReadRequest {
  bytes escrow_name = 1;
}

message EscrowChannelReadResponse {
  bytes escrow_value = 1;
}

message EscrowChannelWriteRequest {
  bytes escrow_name = 1;
  bytes escrow_value = 2;
}

message EscrowChannelWriteResponse {
}

message EscrowChannelDeleteRequest {
  bytes escrow_name = 1;
}

message EscrowChannelDeleteResponse {
}
